### PR TITLE
Qian/query option doc

### PR DESCRIPTION
**Pull request by QianZhu**
- API documentation improvement for queries (table.search)
- a small bug fix for the remote API on create_table
image
image

### CODE REVIEW

File: python/lancedb/context.py [hunk 1]
```diff
@@ -183,7 +195,7 @@ def min_window_size(self, min_window_size: int) -> Contextualizer:
         deprecated_in="0.3.1",
         removed_in="0.4.0",
         current_version=__version__,
-        details="Use the bar function instead",
+        details="Use to_pandas() instead",
```
Comment (eddyxu) [inline python/lancedb/context.py:198]: LMAO

File: python/lancedb/db.py [hunk 2]
```diff
@@ -52,12 +52,24 @@ def create_table(
         ----------
         name: str
             The name of the table.
-        data: list, tuple, dict, pd.DataFrame; optional
-            The data to initialize the table. User must provide at least one of `data` or `schema`.
-        schema: pyarrow.Schema or LanceModel; optional
-            The schema of the table.
+        data: The data to initialize the table, *optional*
+            User must provide at least one of `data` or `schema`.
+            Acceptable types are:
```
Comment (eddyxu) [inline python/lancedb/db.py:57]: Can we make sure this list rendered correctly in the doc

File: python/lancedb/query.py [hunk 3]
```diff
@@ -30,7 +30,40 @@
...
+        top k results to return
+    metric : str
+        the distance metric between a pair of vectors, 
+        
+        can support L2 (default), Cosine and Dot. 
+        [metric definitions][search]
+    columns : Optional[List[str]]
+        which columns to return in the results
+    nprobes : int
+        The number of probes used - optional
```
Comment (eddyxu) [inline python/lancedb/query.py:53]: is this list rendered in the paremeter section , or another section for explaination?

Could you check the rendered doc

File: python/lancedb/query.py [hunk 4]
```diff
@@ -61,6 +94,9 @@ class Query(pydantic.BaseModel):
 
 
 class LanceQueryBuilder(ABC):
+    """LanceDB Query based on specific query type:
```
Comment (eddyxu) [inline python/lancedb/query.py:97]: `Build LanceDB query based on ...`

File: python/lancedb/query.py [hunk 5]
```diff
@@ -133,11 +169,11 @@ def __init__(self, table: "lancedb.table.Table"):
         deprecated_in="0.3.1",
         removed_in="0.4.0",
         current_version=__version__,
-        details="Use the bar function instead",
+        details="Use to_pandas() instead",
     )
     def to_df(self) -> "pd.DataFrame":
         """
-        Deprecated alias for `to_pandas()`. Please use `to_pandas()` instead.
+        *Deprecated alias for `to_pandas()`. Please use `to_pandas()` instead.*
```
Comment (eddyxu) [inline python/lancedb/query.py:176]: Do we need to add a python deprecation warning in the code?

Comment (eddyxu) [reply python/lancedb/query.py:176]: Also should we mention when will we remove this function? (i.e., we will move in in version 1.0) or sth. So it wont be here infinitely .

File: python/lancedb/table.py [hunk 6]
```diff
@@ -149,13 +149,13 @@ class Table(ABC):
...
-        """The [Arrow Schema](https://arrow.apache.org/docs/python/api/datatypes.html#) of
-        this Table
+        """The [Arrow Schema](https://arrow.apache.org/docs/python/api/datatypes.html#)
+        of this Table
 
         """
         raise NotImplementedError
 
-    def to_pandas(self):
+    def to_pandas(self) -> pd.DataFrame:
```
Comment (eddyxu) [inline python/lancedb/table.py:158]: Have we done a round of clean up of pandas dependency, to make pandas optional. Can we check that the code still work if pandas is not installed.

**Review by eddyxu — APPROVED** [review]
A few nits. LGTM overall.

Pending fix on these nits.

File: python/lancedb/query.py (hunk 3 above)
Comment (QianZhu) [reply python/lancedb/query.py:53]: The first screenshot attached presented the rendered doc. Does it look OK?

File: python/lancedb/query.py (hunk 5 above)
Comment (QianZhu) [reply python/lancedb/query.py:176]: we have this decorator:
`@deprecation.deprecated(
        deprecated_in="0.3.1",
        removed_in="0.4.0",
        current_version=__version__,
        details="Use to_pandas() instead",
    )`

File: python/lancedb/db.py (hunk 2 above)
Comment (QianZhu) [reply python/lancedb/db.py:57]: The first screenshot attached presented the rendered doc. Does it look OK?

**Pull request merged**